letsencrypt/client.py

```diff
@@ -30,48 +31,107 @@
 logger = logging.getLogger(__name__)
 
 
-class Client(object):
-    """ACME protocol client.
+def _acme_from_config_key(config, key):
+    # TODO: Allow for other alg types besides RS256
+    return acme_client.Client(new_reg_uri=config.server, key=key,
+                              verify_ssl=(not config.no_verify_ssl))
+
+
+def register(config, account_storage, tos_cb=None):
+    """Register new account with an ACME CA.
+
+    This function takes care of generating fresh private key,
+    registering the account, optionally accepting CA Terms of Service
+    and finally saving the account. It should be called prior to
+    initialization of `Client`, unless account has already been created.
 
-    :ivar network: Network object for sending and receiving messages
-    :type network: :class:`letsencrypt.network.Network`
+    :param .IConfig config: Client configuration.
 
-    :ivar account: Account object used for registration
-    :type account: :class:`letsencrypt.account.Account`
+    :param .AccountStorage account_storage: Account storage where newly
+        registered account will be saved to. Save happens only after TOS
+        acceptance step, so any account private keys or
+        `.RegistrationResource` will not be persisted if `tos_cb`
+        returns ``False``.
 
-    :ivar auth_handler: Object that supports the IAuthenticator interface.
-        auth_handler contains both a dv_authenticator and a
-        continuity_authenticator
-    :type auth_handler: :class:`letsencrypt.auth_handler.AuthHandler`
+    :param tos_cb: If ACME CA requires the user to accept a Terms of
+        Service before registering account, client action is
+        necessary. For example, a CLI tool would prompt the user
+        acceptance. `tos_cb` must be a callable that should accept
+        `.RegistrationResource` and return a `bool`: ``True`` iff the
+        Terms of Service present in the contained
+        `.Registration.terms_of_service` is accepted by the client, and
+        ``False`` otherwise. ``tos_cb`` will be called only if the
+        client acction is necessary, i.e. when ``terms_of_service is not
+        None``. This argument is optional, if not supplied it will
```

Comment:
I think automatic acceptance will get us into trouble.  Hopefully this is intended for testing.